packages/rum-recorder/src/boot/rumRecorderPublicApi.ts

```diff
@@ -24,13 +25,17 @@ export type RumRecorderUserConfiguration = RumRecorderInitConfiguration
 
 const enum RecorderStatus {
   Stopped,
+  IntentToStart,
   Starting,
   Started,
 }
 type RecorderState =
   | {
       status: RecorderStatus.Stopped
     }
+  | {
+      status: RecorderStatus.IntentToStart
```

Comment:
what about adding some comments explaining the difference between these start states?